resources/testharness.js

```diff
@@ -541,7 +541,15 @@ policies and contribution forms [3].
         var test_name = name ? name : test_environment.next_default_test_name();
         properties = properties ? properties : {};
         var test_obj = new Test(test_name, properties);
-        test_obj.step(func, test_obj, test_obj);
+        var value = test_obj.step(func, test_obj, test_obj);
+
+        test_obj.step(function() {
+            assert(value === undefined,
```

Comment:
This is OK, but if I was being picky I might do

```
var msg = "Test object returned a value " + value";
if value.hasOwnProperty("then") {
  msg += " Consider writing promise_test instead";
}
test_obj.set_status(test_obj.ERROR, msg);
test_obj.phase = test_obj.phases.HAS_RESULT;
test_obj.done();
```

Reply:
+1 for @jgraham's suggestion.